editoast/src/views/sub_categories.rs

```diff
@@ -0,0 +1,152 @@
+use axum::Extension;
+use axum::extract::Json;
+use axum::extract::Path;
+use axum::extract::Query;
+use axum::extract::State;
+use axum::response::IntoResponse;
+use editoast_authz as authz;
+use editoast_derive::EditoastError;
+use editoast_models::DbConnectionPoolV2;
+use editoast_schemas::rolling_stock::SubCategory;
+use editoast_schemas::rolling_stock::TrainCategory;
+use serde::Deserialize;
+use serde::Serialize;
+use thiserror::Error;
+use utoipa::IntoParams;
+use utoipa::ToSchema;
+
+use crate::error::Result;
+use crate::views::AuthenticationExt;
+use crate::views::AuthorizationError;
+use crate::views::pagination::PaginationQueryParams;
+use crate::views::pagination::PaginationStats;
+
+crate::routes! {
+    "/sub_category" => {
+        get_sub_categories,
+        create_sub_categories,
+        "/{sub_category_id}" => {
+            delete_sub_category,
+        },
+    },
+}
+
+editoast_common::schemas! {
+    SubCategoryPayload,
+    SubCategoriesCountList,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "sub_categories")]
+pub enum SubCategoryError {
+    #[error(transparent)]
+    #[editoast_error(status = 500)]
+    Database(#[from] editoast_models::model::Error),
+}
+
+#[derive(Debug, Serialize, ToSchema)]
+struct SubCategoriesCountList {
+    results: Vec<SubCategory>,
+    #[serde(flatten)]
+    stats: PaginationStats,
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "sub_categories",
+    responses(
+        (status = 200, description = "The list of all sub categories", body = SubCategoriesCountList),
+    ),
+)]
+async fn get_sub_categories(
+    State(_db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Query(_pagination): Query<PaginationQueryParams<100>>,
+) -> Result<Json<SubCategoriesCountList>> {
+    let authorized = auth
+        .check_roles([authz::Role::OperationalStudies].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+    // TODO implement this endpoint
+    Ok(Json(SubCategoriesCountList {
+        results: vec![],
+        stats: PaginationStats::new(0, 0, 0, 0),
+    }))
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct SubCategoryPayload {
+    pub name: String,
+    pub code: String,
+    pub main_category: TrainCategory,
+    // TODO add color validation
+    pub color: String,
+}
+
+#[utoipa::path(
+    post, path = "",
+    tag = "sub_categories",
+    responses(
+        (status = 200, description = "Create sub categories", body = Vec<SubCategory>),
+    ),
+)]
+async fn create_sub_categories(
+    State(_db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Json(data): Json<Vec<SubCategoryPayload>>,
+) -> Result<Json<Vec<SubCategory>>> {
+    let authorized = auth
+        .check_roles([authz::Role::OperationalStudies].into())
```

Comment:
```suggestion
        .check_roles([authz::Role::Admin].into())
```

Only admins are authorized to create/delete subcategories.